sdk/src/main/java/com/kin/ecosystem/Kin.java

```diff
@@ -192,43 +197,63 @@ public static void login(@NonNull String jwt, KinCallback<Void> loginCallback) {
 	private static void internalLogin(@NonNull final String jwt, final KinCallback<Void> loginCallback) {
 		try {
 			checkInstanceNotNull();
-			if(!AuthRepository.getInstance().isSameUser(jwt)) {
-				logout();
+			@UserLoginState int loginState = AuthRepository.getInstance().getUserLoginState(jwt);
+			switch (loginState) {
+				case UserLoginState.DIFFERENT_USER:
+					logout();
+				case UserLoginState.FIRST:
+					eventLogger.send(UserLoginRequested.create());
+					break;
+				case UserLoginState.SAME_USER:
+					break;
 			}
 			AuthRepository.getInstance().setJWT(jwt);
 		} catch (final ClientException exception) {
 			sendLoginFailed(exception, loginCallback);
 		}
 
-		AuthRepository.getInstance().getAuthToken(new KinCallback<AuthToken>() {
+		AuthRepository.getInstance().getAccountInfo(new KinCallback<AccountInfo>() {
 			@Override
-			public void onResponse(AuthToken authToken) {
+			public void onResponse(AccountInfo accountInfo) {
 				String publicAddress;
 				try {
-					BlockchainSourceImpl.getInstance().loadAccount(authToken.getEcosystemUserID());
+					BlockchainSourceImpl.getInstance().loadAccount(accountInfo.getAuthToken().getEcosystemUserID());
 					publicAddress = BlockchainSourceImpl.getInstance().getPublicAddress();
 				} catch (final BlockchainException exception) {
 					sendLoginFailed(exception, loginCallback);
 					return;
 				}
 				AccountManagerImpl.getInstance().start();
-				AuthRepository.getInstance().updateWalletAddress(publicAddress, new KinCallback<Boolean>() {
-					@Override
-					public void onResponse(Boolean response) {
-						isAccountLoggedIn = true;
-						instance.executorsUtil.mainThread().execute(new Runnable() {
-							@Override
-							public void run() {
-								loginCallback.onResponse(null);
-							}
-						});
-					}
-
-					@Override
-					public void onFailure(KinEcosystemException exception) {
-						sendLoginFailed(exception, loginCallback);
-					}
-				});
+
+				// check server is synced with client, if synced -> skip updateWalletAddress
```

Comment:
consider calling ```AccountManagerImpl.getInstance().start();``` anywhere isAccountLoggedIn is set to true. if it set to false the SDK can't be used anyway. No need to start AccountManagerImpl is not login and it better to start it after we know the user can be login

Reply:
The only place we changing to true is here after getting token